lib/dynamoid/document.rb

```diff
@@ -159,6 +190,28 @@ def update_fields(hash_key_value, range_key_value = nil, attrs = {}, conditions
         end
       end
 
+
+      # Update existing document or create new one.
+      # Similar to `.update_fields`. The only diffirence is creating new document.
+      #
+      # Uses efficient low-level `UpdateItem` API call.
+      # Changes attibutes and loads new document version with one API call.
+      # Doesn't run validations and callbacks. Can make conditional update.
+      # If specified conditions failed - returns `nil`
+      #
+      # @param [Scalar value] partition key
+      # @param [Scalar value] sort key (optional)
+      # @param [Hash] attributes
+      # @param [Hash] conditions
+      #
+      # @return [Dynamoid::Document/nil] updated document
+      #
+      # @example Update document
+      #   Post.update(101, read: true)
+      #
+      # @example Update document
+      #   Post.upsert(101, read: true)
+
```

Comment:
Feel like it's nicer when documentation doesn't have blank line between it and the method.